code/datums/storage.dm

```diff
@@ -0,0 +1,291 @@
+/datum/storage
+	var/atom/parent
+
+	var/max_slots
+	var/max_specific_storage
+	var/max_total_storage
+
+	var/list/is_using = list()
+
+	var/locked = FALSE
+	var/attack_hand_interact = TRUE
+	var/allow_big_nesting = FALSE
+
+	var/atom/movable/screen/storage/boxes //storage display object
+	var/atom/movable/screen/close/closer //close button object
```

Comment:
what happens if these are deleted before the datum?

Reply:
not sure will need to test